Eastern-Research-Group/expert-query
# Feature/242 combine name and id

## Related Issues:
* [EQ-242](https://jira.epa.gov/browse/EQ-242)

## Main Changes:
* Combined "name" and "id" columns throughout the user interface. This was accomplished by creating calculated columns in the database, where necessary.
* Moved column output sorting entirely to the `config/profiles.ts` file.
* Moved filter dependency config to the `config/fields.ts` file. In the current case, this works well as a simplification because columns have similar dependency behavior across profiles.
* Fixed a bug/typo that was causing 'year' values to be dropped when initially passed into the URL.

## Steps To Test:
1. Run the ETL locally.
2. Go to http://localhost:3000/attains/assessmentUnits.
3. Open the _Assessment Unit ID_ field and confirm that the values are displayed in the format _id - name_.
4. Select an ID.
5. Copy the URL, open a new tab, and confirm that the field is populated with the previously selected value.
6. Verify that the _Organization ID_ "source select" drop-down still properly filters the ID values.
7. Execute a query with a newly added ID field, and verify the columns returned include the corresponding "id" and "name" fields but not the "displayName" field.

### N.B.
I didn't remove any indexes from the "id" or "name" fields, but I did remove them from the materialized views. This is because they won't be used (in **v1**) to query column values or to execute queries from the UI, but they'll still benefit queries that are made directly to the server.
